Type Errors

Hi there,

first of all: Thanks a lot for creating trpc :) I just recently stumpled upon it and I'm quite enthusiastic about it's design.

Now to the problem:

I'm trying to use trpc in a pnpm-driven monorepo. When I try to compile the package that's using trpc (both `@trpc/server` and `@trpc/client`), `tsc` fails with:

```
../../../../node_modules/.pnpm/@trpc+server@9.23.4/node_modules/@trpc/server/dist/declarations/src/adapters/node-http/types.d.ts:3:16 - error TS2307: Cannot find module 'qs' or its corresponding type declarations.

3 import qs from 'qs';

[...]

../../../../node_modules/.pnpm/@trpc+server@9.23.4/node_modules/@trpc/server/dist/declarations/src/adapters/standalone.d.ts:5:107 - error TS2344: Type 'IncomingMessage' does not satisfy the constraint 'NodeHTTPRequest'.
  Type 'IncomingMessage' is not assignable to type '{ method?: string; query?: any; body?: unknown; }' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
    Types of property 'method' are incompatible.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

5 export declare type CreateHTTPHandlerOptions<TRouter extends AnyRouter> = NodeHTTPHandlerOptions<TRouter, http.IncomingMessage, http.ServerResponse>;
```

The first one
```
TS2307: Cannot find module 'qs' or its corresponding type declarations.
```

can be fixed by installing `@types/qs`. However, I don't think that should be necessary.

The second one
```
TS2344: Type 'IncomingMessage' does not satisfy the constraint 'NodeHTTPRequest'.
```
isn't explicable to me but at least resembles one of the errors documented in #1022.

This issue was closed by  #1151, which should be included with the trpc version I'm using (9.23.4).

It's possible that this is related to my use of pnpm.

Any hint would be appreciated :) I'll keep you posted, if I find anything myself.

--- a/packages/server/src/adapters/node-http/types.ts
+++ b/packages/server/src/adapters/node-http/types.ts
@@ -1,12 +1,14 @@
 import http from 'http';
-import qs from 'qs';
 import { inferRouterContext } from '../..';
 import { HTTPBaseHandlerOptions } from '../../http/internals/types';
 import { AnyRouter } from '../../router';
 
+interface ParsedQs {
+  [key: string]: undefined | string | string[] | ParsedQs | ParsedQs[];
+}
+
 export type NodeHTTPRequest = http.IncomingMessage & {
-  method?: string;
-  query?: qs.ParsedQs;
+  query?: ParsedQs;
   body?: unknown;
 };
 export type NodeHTTPResponse = http.ServerResponse;

